### PR TITLE
ENH: psd topomap improvements

### LINKED ISSUE #7586: ENH: proposed improvements to plot_psd_topomap
Some things I wished I was able to do when using `epochs.plot_psd_topomap` yesterday, and some ideas about how to implement them:

1. **automatic joint determination of vlims.** copy over the API from #6922 that allows `vlim='joint'` for multi-topomap figures

2. **logarithmic colormap.**  SSVEP carrier frequency ought to have more power than oddball frequency, by virtue of the experimental design (fewer oddball stimulus exposures).  A logarithmic colormap might help make the carrier & oddball spikes both visible given the same vmin/vmax.  In addition to the sharp SSVEP spikes, the data I'm currently working with has a large broad bump in the spectrum, with power roughly an order of magnitude greater than the spikes (probably a muscle artifact).  A logarithmic colormap would make it possible to plot that in the same figure with the spikes (useful during exploratory analysis).

3. **make it easier to plot a specific frequency bin.**  Because SSVEP leads to very focal spikes in the spectrum, the current API of `bands=[(band1_lower_edge, band1_upper_edge, 'subplot title'),]` is a inconvenient. It would be nice to be able to pass a specific single frequency and have it not fail. Possible API ideas:
    - pass in a single frequency, get back a topomap of the single bin nearest to that frequency
    - pass in a single frequency and a bandwidth
    - pass in a single frequency and an integer number of bins

For comparison, currently to get a single bin topomap you have to compute the bin spacing and pick bin lower/upper edges that are ± half a bin width from the target frequency. But knowing the bin spacing requires computing the PSD first before plotting (and the PSD then gets computed again within the plotting function, which is not very efficient).

### PR DESCRIPTION
**Pull request by drammock**
closes #7586 

- [x] allow `epochs.plot_psd_topomap(..., vlim='joint')`
- [x] ~~allow logarithmic colormap~~ YAGNI: already a parameter `dB=True`
- [x] make single-frequency-bin band selection easy
- [x] test `vlim='joint'`
- [x] ~~test logarithmic colormap~~ YAGNI
- [x] test single-bin selection
- [x] update example/tutorial to demo new features
- [x] add `cbar_fmt='auto'` to show fewer decimals for dB scale
- [x] don't default to RdBu_r colormap: power is one-sided

### CODE REVIEW

**Comment by drammock** [comment]
rendered docs:

- [updated tutorial section](https://19434-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/epochs/plot_20_visualize_epochs.html#plotting-the-power-spectrum-of-epochs)
- [method docstring](https://19434-1301584-gh.circle-artifacts.com/0/dev/generated/mne.Epochs.html?highlight=plot_psd_topomap#mne.Epochs.plot_psd_topomap)

**Comment by drammock** [comment]
Ready for review.

@larsoner FYI, the solution for single-bin frequency bands ended up being a little different than discussed in #7586:  I decided that changing the existing API from `(lower_edge, upper_edge, title)` to `(center, bandwidth, title)` would be too disruptive to existing user code. So the API now accepts `(lower_edge, upper_edge, title)` (old way) or `(single_frequency, title)` (single-bin).

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -2001,15 +1949,34 @@ def plot_psds_topomap(
...
+    if vmin is not None or vmax is not None:
+        if vlim == (None, None):
+            msg = ('Since you didn\'t specify "vlim", your provided values of '
+                   '"vmin" and "vmax" will be used.')
+            vlim = (vmin, vmax)
+        else:
+            msg = ('Your provided values for "vlim" will be used, and "vmin" '
+                   'and "vmax" will be ignored.')
+        warn('"vmin" and "vmax" are deprecated in favor of a single argument '
+             '"vlim". ' + msg, DeprecationWarning)
```
Comment (larsoner) [inline mne/viz/topomap.py:1964]: "and will be removed in 0.21" somewhere in there. Usually we just say "vmin and vmax are deprecated and will be removed in 0.21, use vlim instead"

File: doc/changes/latest.inc [hunk 2]
```diff
@@ -19,6 +19,8 @@ Changelog
 
 - Add support for signals in mV for :func:`mne.io.read_raw_brainvision` by `Clemens Brunner`_
 
+- :meth:`mne.Epochs.plot_psd_topomap`, :func:`mne.viz.plot_epochs_psd_topomap`, and :func:`mne.viz.pot_psds_topomap` now allow joint colorbar limits across subplots, by `Daniel McCloy`_.
```
Comment (larsoner) [inline doc/changes/latest.inc:22]:
```
changes/latest.inc:22: WARNING: py:func reference target not found: mne.viz.plot_epochs_psd_topomap
changes/latest.inc:22: WARNING: py:func reference target not found: mne.viz.pot_psds_topomap
```

File: tutorials/epochs/plot_20_visualize_epochs.py [hunk 3]
```diff
@@ -131,6 +131,32 @@
 epochs['auditory'].plot_psd(picks='eeg')
 
 ###############################################################################
+# It is also possible to plot spectral estimates across sensors as a scalp
+# topography, using :meth:`~mne.Epochs.plot_psd_topomap`. The default
+# parameters will plot five frequency bands (δ, θ, α, β, γ), and will plot
+# the power estimates in decibels:
+
+epochs['visual/right'].plot_psd_topomap()
```
Comment (agramfort) [inline tutorials/epochs/plot_20_visualize_epochs.py:139]: I think we should also give the intervals used to define the bands in the topo subtitles.
There is perfect consensus on what are the "correct" bands (for good reasons). It's just a good default but it should be explicit.

Comment (agramfort) [reply tutorials/epochs/plot_20_visualize_epochs.py:139]: also what do you think about showing units on the colorbar following the way it's done with plot_psd?

**Comment by drammock** [comment]
@agramfort comments addressed.  To include the units I had to move the colorbar title to be a ylabel, otherwise it collided with the subplot (band name) titles.

**Comment by agramfort** [comment]
looks great

**Comment by agramfort** [comment]
here: https://19484-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/epochs/plot_20_visualize_epochs.html#sphx-glr-auto-tutorials-epochs-plot-20-visualize-epochs-py

**Comment by larsoner** [comment]
Thanks @drammock

**Pull request merged**